### ISSUE TITLE
How to make edge features undirected?

### ISSUE THREAD

**Issue by LindaSt** [rusty1s/pytorch_geometric#914]

**Comment by rusty1s** [comment]
How about this?
```python
row, col = edge_index
new_row = torch.cat([row, col], dim=0)
new_col = torch.cat([col, row], dim=0)
new_edge_index = torch.stack([new_row, new_col], dim=0)
new_edge_attr = torch.cat([edge_attr, edge_attr], dim=0)
```

**Comment by LindaSt** [comment]
Hi! Thanks! Somehow most of my question text was not posted. I was actually asking if there was a implemented funciton, as I saw there is one for the edge_index matrix.

**Comment by rusty1s** [comment]
No, there is not, but I agree that it would be cool to have :)

**Comment by VokCow** [comment]
Hi!

I see that the `to_undirected` function not only uses `torch.cat` and `torch.stack`, but also  `torch_sparse.coalesce`. 

In your opinion, what would be the best way to obtain the elements in `new_edge_attr` corresponding to the edge ordering in the coalesced `new_edge_index`?

Thanks in advance,

Victor.

**Comment by rusty1s** [comment]
This is not really defined in case of edge attributes. If you are sure that reverse edges do not result in duplicated edges, you can simply duplicate your `edge_attr` matrix: `torch.cat([edge_attr, edge_attr], dim=0)`. In case this does not hold, you need some way to unify duplicate edges into a single representation (doable via `torch-scatter`). You can pick any method here, e.g., add, mean or max.
